### DISCUSSION TITLE
How to set custom value for start axis

### DISCUSSION THREAD

**Discussion by bill-liu-0x63** [patrykandpatrick/vico#755] [Questions]
### Question

Hi, I am using Kotlin jetpack compose to build the app. Currently, the values in start axis are auto-scaled related to the data I put into the chart. I want the values in start axis to be fixed, like: [0 * goal, 0.5 * goal, 1 * goal, 1.5 * goal, 2 * goal] (goal is a float value I provide to the composable). 

my code: https://gist.github.com/D4r3E-1v1l/fbd77a6e4fb04ab9c7b5b7cc1a766186

### Vico version(s)

1.7.1

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, @D4r3E-1v1l. This can be done easily in Vico 2:

```kt
itemPlacer = VerticalAxis.ItemPlacer.step({ goal / 2 })
```

In Vico 1, it's less straightforward:

- Versions 1.9.0 to 1.15.0 have the `AxisItemPlacer.Vertical` API. However, there's only one factory function, and it's count-based, not step-based. You can calculate the required number of labels, but this will work only if the maximum _y_ value is a multiple of `goal`, and it may not behave perfectly with `ChartEntryModelProducer` due to its asynchronous processing. A more robust solution for these versions is to create a custom `AxisItemPlacer.Vertical` implementation that replicates the behavior of Vico 2's `VerticalAxis.ItemPlacer.step`.
- In version 1.7.1, `VerticalAxis` has a `maxLabelCount` property. It's a similar story as with the single factory function mentioned in the point above. In this version, to implement the desired behavior more robustly, you'd have to create a custom `Axis` implementation.

With this in mind, your best choice is most likely to update to Vico 2.
